### ISSUE TITLE
Failed to start SSH session: Unable to exchange encryption keys

### ISSUE THREAD

**Issue by ondj** [rust-lang/cargo#5639]
Hello, 

I am using Cargo with private repository on my server. But when I run `cargo check` on my Windows machine (Windows 10 1803), this output is returned:

```
DEBUG 2018-06-19T08:47:01Z: cargo::sources::git::utils: doing a fetch for ssh://XXX
DEBUG 2018-06-19T08:47:01Z: cargo::sources::git::utils: initiating fetch of refs/heads/*:refs/heads/* from ssh://XXX
DEBUG 2018-06-19T08:47:01Z: cargo::sources::git::utils: fetch failed: failed to start SSH session: Unable to exchange encryption keys; class=Ssh (23)
 INFO 2018-06-19T08:47:01Z: cargo::util::job::imp: found 0 remaining processes
DEBUG 2018-06-19T08:47:01Z: cargo: exit_with_error; err=CliError { error: Some(Error { code: -1, klass: 23, message: "failed to start SSH session: Unable to exchange encryption keys" }

failed to fetch into C:\Users\XXX\.cargo\git\db\XXX

Unable to update ssh://XXX

failed to load source for a dependency on `XXX`), unknown: false, exit_code: 101 }
error: failed to load source for a dependency on `XXX`

Caused by:
  Unable to update ssh://XXX

Caused by:
  failed to fetch into C:\Users\XXX\.cargo\git\db\XXX

Caused by:
  failed to start SSH session: Unable to exchange encryption keys; class=Ssh (23)
```

In log on the server side is this error message:

```
no matching cipher found. Their offer: aes256-cbc,rijndael-cbc@lysator.liu.se,aes192-cbc,aes128-cbc,arcfour128,arcfour,3des-cbc [preauth]
```

All offered ciphers are disabled on the server side, because are considered not safe. I found this comment https://github.com/rust-lang/cargo/issues/3492#issuecomment-312461080, thats probably explain this behavior. But from code of libssh2 library, that is used by `git2` crate, at least `aes256-ctr` is supported by wincng backend (https://github.com/libssh2/libssh2/blob/master/src/wincng.h#L318), but not send as supported cipher by cargo.

**Comment by matklad** [comment]
Classifying it as a bug. I personally have little experience with SSH on windows, so I won't be able to track this down myself.

**Comment by ondj** [comment]
I tried to use git2 to solve that problem and when I used this `Cargo.toml` (version used in version 1.27.0), the problem was there:

```
[dependencies]
git2 = "=0.7.0"
libgit2-sys = "=0.7.1"
libssh2-sys = "=0.2.6"
```
but update `libssh2-sys` crate to version 0.2.7, that was released May 3, 2018, helped and everything worked then. 

But this repo doesnt contains `Cargo.lock` and latest nightly Cargo from 2018-06-07 still contains libssh2-sys in version 0.2.6. So how it is possible to update libssh2-sys crate to latest version?
